### PR TITLE
Remove c++11 compile flags so that xerces-c can be used on projects t…

### PR DESCRIPTION
**Pull request by jsharpe**
…hat use c++98

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
we usually don't want to "go backwards" on C++ standards but we'll see what @fxcoudert thinks

**Comment by jsharpe** [comment]
Given it compiles in c++98 mode shouldn't that be the default as it is most compatible? With the current formula it can only be used with at least a c++11 compiler. The project I'm trying to use it against is c++98 only so I currently can't use this formula to provide this dependency due to it exposing a c++11 only type (char16_t) in its header. This fix uses the more compatible uint16_t which should be forward compatible for C++11 projects as well.

File: Formula/xerces-c.rb [hunk 1]
```diff
@@ -13,13 +13,9 @@ class XercesC < Formula
...
   depends_on "cmake" => :build
 
-  needs :cxx11
-
   def install
-    ENV.cxx11
-
     mkdir "build" do
-      system "cmake", "..", *std_cmake_args
+      system "cmake", "..", "-Dxmlch-type=uint16_t", *std_cmake_args
```
Comment (MikeMcQuaid) [inline Formula/xerces-c.rb]: Why is the `xmlch-type` required?

**Comment by MikeMcQuaid** [comment]
> we usually don't want to "go backwards" on C++ standards but we'll see what @fxcoudert thinks

Agreed 👍 

> Given it compiles in c++98 mode shouldn't that be the default as it is most compatible?

Not necessarily. If you're using a c++11 project does your change make it any less useful i.e. remove functionality?

**Comment by fxcoudert** [comment]
We have the following formulas that depend on `xerces-c` and use C++11 explicitly:
`blahtexml dbxml enigma opensaml shibboleth-sp xalan-c xml-security-c xml-tooling-c xqilla xsd zorba`

macOS system compilers are C++11 compliant, and more and more code is relying on C++11. So I don't think moving back is a good choice for us.

**Comment by ilovezfs** [comment]
@jsharpe for some context, we started enforcing C++11 for all of the xerces-c reverse dependencies in https://github.com/Homebrew/homebrew-core/pull/17371. You may want to keep C++98 version in [your own tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md#how-to-create-and-maintain-a-tap). Thank you for this PR nonetheless.

**Pull request closed without merging**